com.unity.render-pipelines.core/Editor/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,181 @@
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+namespace UnityEditor.Rendering
+{
+	/// <summary>
+	/// Material upgrader and relevant utilities for SpeedTree8.
+	/// </summary>
+	public class SpeedTree8MaterialUpgrader : MaterialUpgrader
+    {
+        private enum WindQuality
+        {
+            None = 0,
+            Fastest,
+            Fast,
+            Better,
+            Best,
+            Palm,
+            Count
+        }
+
+        private static string[] WindQualityString =
+        {
+            "_WINDQUALITY_NONE",
+            "_WINDQUALITY_FASTEST",
+            "_WINDQUALITY_FAST",
+            "_WINDQUALITY_BETTER",
+            "_WINDQUALITY_BEST",
+            "_WINDQUALITY_PALM"
+        };
+
+        /// <summary>
+        /// Creates a material upgrader with only the renames in common between HD and Universal.
+        /// </summary>
+        /// <param name="sourceShaderName">Original ST8 shader name.</param>
+        /// <param name="destShaderName">New ST8 shader name.</param>
+        /// <param name="finalizer">A delegate that postprocesses the material as needed by the render pipeline.</param>
+        public SpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName, MaterialFinalizer finalizer = null)
+        {
+            RenameShader(sourceShaderName, destShaderName, finalizer);
+            RenameFloat("_WindQuality", "_WINDQUALITY");
+            RenameFloat("_TwoSided", "_CullMode"); // Currently only used in HD. Update this once URP per-material cullmode is enabled via shadergraph. 
+        }
+
+        private static void ImportNewSpeedTree8Material(Material mat, int windQuality, bool isBillboard)
+        {
+            int cullmode = 0;
+            mat.SetFloat("_WINDQUALITY", windQuality);
+            if (isBillboard)
+            {
+                mat.EnableKeyword("EFFECT_BILLBOARD");
+                cullmode = 2;
+            }
+            if (mat.HasProperty("_CullMode"))
+                mat.SetFloat("_CullMode", cullmode);
+        }
+
+        /// <summary>
+        /// Postprocess materials when importing a SpeedTree8 asset. Call from OnPostprocessSpeedTree in a MaterialPostprocessor.
```

Comment:
The vendor seems to prefer SpeedTree 8 to SpeedTree8. Also potentially confusing because this document refers to ST8 in at least one other summary.

Reply:
Postprocesses materials while you are importing a SpeedTree 8 asset. Call from OnPostprocessSpeedTree in a MaterialPostprocessor.

(need to make sure that user knows how to call things from a MaterialPostprocessor. A link  or more info is likely needed).

Reply:
How do you add links in these comment-style docs? I think a link to MaterialPostprocessor documentation would be sufficient here.

That documentation might not exist, but it shouldn't be added in this PR.

Reply:
Not sure we can link, actually, and didn't want to add full doc for that to this PR, just a "see the doc for" kind of thing. Don't seem to find that doc, so I'll create a ticket. What about the other proposed changes?

Reply:
I copy-pasted the rewrite you proposed in c29ee75